lib/balance-tx/cardano-balance-tx.cabal

```diff
@@ -67,30 +65,31 @@ library
     , generics-sop
     , int-cast
     , lens
+    , MonadRandom
     , monoid-subclasses
     , ouroboros-consensus
     , ouroboros-consensus-cardano
     , pretty-simple
+    , QuickCheck
     , random
     , serialise
     , std-gen-seed
     , text
     , transformers
   exposed-modules:
-    Cardano.Tx.Balance
-    Cardano.Tx.Balance.Internal.CoinSelection
-    Cardano.Tx.Balance.Internal.CoinSelection.Gen
     Cardano.Wallet.Shelley.Compatibility.Ledger
-    Cardano.Wallet.Write.ProtocolParameters
-    Cardano.Wallet.Write.Tx
-    Cardano.Wallet.Write.Tx.Balance
-    Cardano.Wallet.Write.Tx.Balance.TokenBundleSize
-    Cardano.Wallet.Write.Tx.Gen
-    Cardano.Wallet.Write.Tx.Redeemers
-    Cardano.Wallet.Write.Tx.Sign
-    Cardano.Wallet.Write.Tx.SizeEstimation
-    Cardano.Wallet.Write.Tx.TimeTranslation
-    Cardano.Wallet.Write.UTxOAssumptions
+    Cardano.Write.ProtocolParameters
+    Cardano.Write.Tx
+    Cardano.Write.Tx.Balance
+    Cardano.Write.Tx.Balance.TokenBundleSize
+    Cardano.Write.Tx.Gen
+    Cardano.Write.Tx.Redeemers
+    Cardano.Write.Tx.Sign
+    Cardano.Write.Tx.SizeEstimation
+    Cardano.Write.Tx.TimeTranslation
+    Cardano.Write.UTxOAssumptions
+    Internal.Cardano.Write.Tx.Balance.CoinSelection
+    Internal.Cardano.Write.Tx.Balance.CoinSelection.Gen
```

Comment:
I'm guessing this is the beginning of a pattern? I'm not fully convinced, but happy to see where it takes us and/or to discuss further when convenient.

Reply:
> Could we discuss this first?

Sure. Happy to discuss all options!